llvm/lib/ObjectYAML/DXContainerYAML.cpp

```diff
@@ -347,6 +353,11 @@ void DXContainerYAML::PSVInfo::mapInfoForVersion(yaml::IO &IO) {
   IO.mapRequired("NumThreadsX", Info.NumThreadsX);
   IO.mapRequired("NumThreadsY", Info.NumThreadsY);
   IO.mapRequired("NumThreadsZ", Info.NumThreadsZ);
+
+  if (Version == 2)
+    return;
+
+  IO.mapRequired("EntryFunctionName", Info.EntryFunctionName);
```

Comment:
In the YAML this should be a string not an integer. The Integer is the offset into the string table, for the textual dump we want to read this out into a string and be able to rebuild the string table.

The `DirectX::PSVRuntimeInfo` object has a method `getStringTable()`, which returns a StringRef that represents the full string table (just a bunch of packed null-terminated strings). The new field is an offset into that table for where the entry function name is stored.

In `dxcontainer2yaml.cpp` we'll want to translate that offset to a string that is stored in the `DXContainerYAML::PSVInfo` object, and we'll need to pass the string into the MC writer to add it into the string table builder.

Reply:
Lastet commit updates to use a string in the yaml and writes the value to a string table.  There was a bit of refactoring that had to be done because I needed the offset of the entrypoint to be written into the runtime info before it was written out.  So I am sure more cleanups are necessary, but I believe this is heading in the correct direction.